src/cpp/src/continuous_batching_impl.cpp

```diff
@@ -752,7 +752,7 @@ void ContinuousBatchingPipeline::ContinuousBatchingImpl::_fill_prompt_log_probs(
         }
         currently_processed_tokens += output_seq_len * num_running_sequences;
         // For max_new_tokens == 0, we don't reach sampling so need to notify handle separately
-        if(sequence_group->get_sampling_parameters().max_new_tokens == 0) {
+        if(sequence_group->get_sampling_parameters().get_max_new_tokens(sequence_group->get_prompt_len()) == 0) {
```

Comment:
```suggestion
        if(sequence_group->get_max_new_tokens() == 0) {
```

`SequenceGroup` now has `get_max_new_tokens` method: https://github.com/openvinotoolkit/openvino.genai/blob/master/src/cpp/src/sequence_group.hpp#L702